src/launchpad/service.py

```diff
@@ -364,31 +401,36 @@ def _download_artifact_to_temp_file(
                 self._safe_cleanup(temp_file, "temporary file")
             raise
 
-    def _prepare_update_data(self, app_info: AppleAppInfo | BaseAppInfo, artifact: Any) -> Dict[str, Any]:
-        """Prepare update data based on app platform and artifact type."""
+    def _prepare_update_data(self, app_info: AppleAppInfo | BaseAppInfo, artifact: Artifact) -> Dict[str, Any]:
+        def _get_artifact_type(artifact: Artifact) -> ArtifactType:
+            if isinstance(artifact, ZippedXCArchive):
+                return ArtifactType.XCARCHIVE
+            elif isinstance(artifact, (AAB, ZippedAAB)):
+                return ArtifactType.AAB
+            else:
+                return ArtifactType.APK
+
+        update_data = {
+            "app_name": app_info.name,
+            "app_id": app_info.app_id,
+            "build_version": app_info.version,
+            "build_number": (int(app_info.build) if app_info.build.isdigit() else None),
+            "artifact_type": _get_artifact_type(artifact).value,
+        }
+
         if isinstance(app_info, AppleAppInfo):
             # TODO: add "date_built" field once exposed in 'AppleAppInfo'
-            return {
-                "build_version": app_info.version,
-                "build_number": (int(app_info.build) if str(app_info.build).isdigit() else app_info.build),
-                "artifact_type": ArtifactType.XCARCHIVE.value,
-                "apple_app_info": {
-                    "is_simulator": app_info.is_simulator,
-                    "codesigning_type": app_info.codesigning_type,
-                    "profile_name": app_info.profile_name,
-                    "is_code_signature_valid": app_info.is_code_signature_valid,
-                    "code_signature_errors": app_info.code_signature_errors,
-                },
-            }
-        else:
-            artifact_type = ArtifactType.AAB if isinstance(artifact, (AAB, ZippedAAB)) else ArtifactType.APK
-            # TODO: add "date_built" and custom android fields
-            return {
-                "build_version": app_info.version,
-                "build_number": (int(app_info.build) if app_info.build.isdigit() else None),
-                "artifact_type": artifact_type.value,
+            update_data["apple_app_info"] = {
```

Comment:
Are we able to add types to this service class? I'm hoping we can use pydantic serialization instead of these brittle dictionaries. Don't have to do it as part of this PR.